buildkite/scripts/build_python_client.sh

```diff
@@ -1,3 +1,62 @@
 #!/bin/bash
 
- make build-python-apiclient
+set -e
+
+setup_environment_variables() {
+  export PYPI_TOKEN=$(buildkite-agent secret get PYPI_TOKEN)
+}
+
+generate_swagger() {
+  local path_to_project=$(git rev-parse --show-toplevel)
+  local swagger_dir="${path_to_project}/pkg/swagger"
+  local webui_path="${path_to_project}/webui/lib/api/schema"
+  local docs_path="${path_to_project}/docs"
```

Comment:
since swagger is part of our repo, should we generate it on the fly when building the python client, or should we use the swagger definition already in our repo? I understand they "should" be kept up to date, but if there was an issue and we didn't keep them up to date, this might cause an unpredictable behaviour, or what do you think?

what we should do actually is add a verification to our build pipeline to generate swagger and fail if the generated one doesn't match the one in our repo, which will force anyone doing changes to the APIs or models to make sure to regenerate swagger

Reply:
Maybe I am missing something. If the generated one does not match the one in repo then it means that the swagger  was not generated, even though source code had changed. It is too easy to forget to commit  updated swagger, given the fact that it is generated from GoLang Code.

There are 2 options ideally,

-  As part of your git commit hook you always run swagger generation, as a result no one forgets and they get reviews as part of PR Review
- Or else, you always generate the swagger when needed and don't at all commit to your database.

Reply:
> As part of your git commit hook you always run swagger generation, as a result no one forgets and they get reviews as part of PR Review

Exactly. A hook or a dedicated task in buildkite build pipeline. A pre-commit hook can be an overkill when running locally as I run those before each commit. In the build pipeline can be a better fit

Reply:
Sorry I missed this. Makes sense. I ll add a dedicated task.